src/utils/docsParser.ts

```diff
@@ -226,29 +244,37 @@ export class LazyModuleDocumentation implements IModuleMetadata {
     this.name = name;
   }
 
-  public get rawDocumentation(): Record<string, unknown> {
+  public get rawDocumentationFragments(): Map<string, Record<string, unknown>> {
     if (!this._contents) {
+      this._contents = new Map<string, Record<string, unknown>>();
       const contents = fs.readFileSync(this.source, { encoding: 'utf8' });
-      const m = LazyModuleDocumentation.docsRegex.exec(contents);
-      if (m && m.groups && m.groups.doc && m.groups.pre) {
-        // determine documentation start/end lines for definition provider
-        let startLine = contents.substr(0, m.index).match(/\n/g)?.length || 0;
-        startLine += m.groups.pre.match(/\n/g)?.length || 0;
-        const endLine = startLine + (m.groups.doc.match(/\n/g)?.length || 0);
-        this.sourceLineRange = [startLine, endLine];
-
-        const document = parseDocument(m.groups.doc);
-        // There's about 20 modules (out of ~3200) in Ansible 2.9 libs that contain YAML syntax errors
-        // Still, document.toJSON() works on them
-        this._contents = document.toJSON();
-        this.errors = document.errors;
+      let m;
+      while ((m = LazyModuleDocumentation.docsRegex.exec(contents)) !== null) {
```

Comment:
The whole loop body is under a conditional block. It seems like if the conditional clause evaluates to `false`, this will end up being an infinite loop.

Reply:
@tomaciazek Can you please make a follow-up to address this?

Reply:
There's nothing to fix - this is how regex works in JavaScript, and this particular line is straight from [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#finding_successive_matches). This loop will always terminate unless it's content interacts (in a very particular way) directly with `LazyModuleDocumentation.docsRegex`, which it doesn't.